tools/history_tools.py

```
@@ -12,16 +12,29 @@ def __init__(self):
         
         Args:
-            limit (int): The maximum number of messages to return. Defaults to 2000.
+            limit (int): The maximum number of messages to return. 
+                         Must be between 1 and 10000. Defaults to 2000.
             
         Returns:
             str: The chat history as a string, or a message indicating no history found.
         """
+        # Validate and clamp limit to safe range
+        if not isinstance(limit, int) or limit < self.MIN_LIMIT:
+            logger.warning(f"[HistoryTools] Invalid limit {limit}, using minimum {self.MIN_LIMIT}")
```

Ruff (0.15.2)

[warning] Logging statement uses f-string

(G004)